src/tribol/common/ExecModel.hpp

```diff
@@ -0,0 +1,312 @@
+// Copyright (c) 2017-2023, Lawrence Livermore National Security, LLC and
+// other Tribol Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (MIT)
+ 
+#ifndef SRC_COMMON_EXECMODEL_HPP_
+#define SRC_COMMON_EXECMODEL_HPP_
+
+// Tribol includes
+#include "tribol/common/BasicTypes.hpp"
+
+// Axom includes
+#include "axom/core/memory_management.hpp"
+#include "axom/slic.hpp"
+
+namespace tribol
+{
+
+enum class MemorySpace
+{
+  Dynamic,
+  Host,
+#ifdef TRIBOL_USE_UMPIRE
+  Device,
+  Unified
+#endif
+};
+
+enum class ExecutionMode
```

Comment:
Yes, the ifdefs should be indented.  I got rid of the TRIBOL_USE_RAJA check, since RAJA is required for anything besides sequential.  See checks in LoopExec.hpp